docs/source/contributing/developing.adoc

```diff
@@ -274,12 +274,26 @@ RUN_BEFORE_FEATURE::
 The `RUN_BEFORE_FEATURE` parameter specifies {project} commands to be run before each feature.
 This provides the ability to run integration tests against a {project} instance which is not in its default state.
 When multiple commands are specified, they must be delimited by a semicolon.
-For example, tests can be run against a stopped {project} instance with the image caching option turned on by running the following:
+For example, tests can be run against a stopped {project} instance with the image caching option turned on by running the following: 
 
 ----
 $ make integration_all RUN_BEFORE_FEATURE="start; stop; config set image-caching true"
 ----
 
+TEST_WITH_SHELL::
+
+About:::
+Majority of integration tests in Minishift passes command arguments directly to `minishift` or `oc` binaries with usage of Go's `os/exec` package.
+However some integration tests first starts an instance of shell, into which the tests later passes the commands.
+This attitude allows to test `minishift docker-env` or `minishift oc-env` commands, which are dependent on which kind of shell they are being executed in.
+The steps using this approach have the `shell` keyword in their definition, for example: `When executing in shell "minishift oc-env" succeeds`.
+
+Usage:::
+`TEST_WITH_SHELL` is an optional parameter.
+If nothing is passed with `TEST_WITH_SHELL`, then the shell type is being detected automatically: for Windows the `PowerShell`, while on macOS and Linux the `bash` will be used.
+To run the tests with different shells, TEST_WITH_SHELL parameter can be used, for example run: `minishift integration_all TEST_WITH_SHELL=cmd` to execute the tests with Windows' `Command Prompt`.
```

Comment:
Note: I assume that `minishift integration_all` was a typo, and `make integration_all` was intended.

```adoc
The `TEST_WITH_SHELL` parameter can be used to run the tests with different shells.
To run the tests using the Windows Command Prompt, for example, run `make integration_all TEST_WITH_SHELL=cmd`.
```